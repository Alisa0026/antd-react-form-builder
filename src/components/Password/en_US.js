export default {
  FormBuilderPassword: {
    "label": "verify password",
    "checkErrorMsg": "The two passwords are not the same",
  },
}
