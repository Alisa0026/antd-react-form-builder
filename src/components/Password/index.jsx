import React from 'react'
import _ from 'lodash'
import moment from 'moment'
import BasicItem from '../BasicItem'
import Input from 'antd/lib/input'
import FormItemComponentDecorator from '../../decorator/FormItemComponent'
import localeDecorator from "../../decorator/Locale"
import localeText from './zh_CN'

function component(BasicItemComponent){
  @FormItemComponentDecorator
  @localeDecorator
  class Password extends React.Component {
    static propTypes = {
      min: React.PropTypes.oneOfType([
        React.PropTypes.string,
        React.PropTypes.number,
      ]),
      max: React.PropTypes.oneOfType([
        React.PropTypes.string,
        React.PropTypes.number,
      ]),
      rePassword: React.PropTypes.bool,
      onlyLetterAndNumber: React.PropTypes.bool,
    }

    render(){
      let { 
        locale,
        key,
        rePassword,
        onlyLetterAndNumber=true,
        ...other 
      } = this.props;
      //console.debug(this.context)
      locale = this.getLocale(localeText,"FormBuilderPassword"); 

      other.targetComponent = Input;
      other.type = "password";
      this.propsAdapter(other);
      if(onlyLetterAndNumber){
        other.rules.push({
          validator(rule, value, callback, source, options) {
            var errors = [];
            var pass = new RegExp("^[A-Za-z]+[0-9]+[A-Za-z0-9]*|[0-9]+[A-Za-z]+[A-Za-z0-9]*$").test(value);
            if(!pass && value != ""){
              errors.push({
                message: locale.formatErrorMsg,
              })
            }
            callback(errors);
          }
        })
      }
      if(rePassword){
        var reProps = _.cloneDeep(other);
        reProps.rules = [];
        reProps.min = undefined;
        reProps.max = undefined;
        reProps.formItemProps.label = locale.reLabel, 
        reProps.name = "re-" + reProps.name;
        reProps.rules.push(
          {
            validator(rule, value, callback, source, options) {
              var errors = [];
              var password_value = document.getElementsByName(other.name)[0].value;
              if(password_value !== value){
                errors.push({
                  message: locale.checkErrorMsg,
                })
              }
              callback(errors);
            }
          }
        )
      }
      return (
        <div key={ key } className="password-con">
          <BasicItemComponent { ...other }/>
          {
            rePassword &&
            <BasicItemComponent { ...reProps } />
          }
        </div>
      )
    }
    
  }
  return Password;
}

export default component(BasicItem) 




