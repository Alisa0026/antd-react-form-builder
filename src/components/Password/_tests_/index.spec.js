import React from 'react'
import { shallow,mount,render } from 'enzyme'
import sinon from 'sinon'
import LocaleProvider from "antd/lib/locale-provider"
import { 
  testCommonProps, 
  getAntdFormComponentMount,
  getAntdFormItemMount,
} from '../../../../tests/common'
import Password from '../index'

describe('Password',function(){
  sinon.stub(console,"warn");
  it('renders correctly', () => {
    const wrapper = render(
      <Password 
        required
        className="test"
        label="name"
        name="test"
        rePassword={ true }
        min={ 5 }
        max={ 12 }
        value="55"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  testCommonProps(Password,"password"); 
  it("min='5' should show success state",function(){
    const wrapper = mount(
      <Password 
        name="test"
        value="111ddd11"
        onlyLetterAndNumber={ false }
        min={ 5 }
      />
    );
    const antdFormItemMount = getAntdFormItemMount(wrapper,"password");
    expect(antdFormItemMount.childAt(0).childAt(0).props().className).toEqual("ant-form-item-control has-success");
  })
  it("min='5' should show error state",function(){
    const wrapper = mount(
      <Password 
        name="test"
        value="11dd"
        onlyLetterAndNumber={ false }
        min={ 6 }
      />
    );
    const antdFormItemMount = getAntdFormItemMount(wrapper,"password");
    expect(antdFormItemMount.childAt(0).childAt(0).props().className).toEqual("ant-form-item-control has-error");
  })
  it("max='5' should show success state",function(){
    const wrapper = mount(
      <Password 
        name="test"
        value="11d1"
        onlyLetterAndNumber={ false }
        max={ 5 }
      />
    );
    const antdFormItemMount = getAntdFormItemMount(wrapper,"password");
    expect(antdFormItemMount.childAt(0).childAt(0).props().className).toEqual("ant-form-item-control has-success");
  })
  it("max='5' should show error state",function(){
    const wrapper = mount(
      <Password 
        name="test"
        onlyLetterAndNumber={ false }
        value="11d11111"
        max={ 5 }
      />
    );
    const antdFormItemMount = getAntdFormItemMount(wrapper,"password");
    expect(antdFormItemMount.childAt(0).childAt(0).props().className).toEqual("ant-form-item-control has-error");
  })
  it("min='2' && max='5' should show success state",function(){
    const wrapper = mount(
      <Password 
        name="test"
        value="1d1"
        onlyLetterAndNumber={ false }
        min={ 2 }
        max={ 5 }
      />
    );
    const antdFormItemMount = getAntdFormItemMount(wrapper,"password");
    expect(antdFormItemMount.childAt(0).childAt(0).props().className).toEqual("ant-form-item-control has-success");
  })
  it("min='2' && max='5' && value='1111111111' should show error state",function(){
    const wrapper = mount(
      <Password 
        name="test"
        value="11111111d11"
        onlyLetterAndNumber={ false }
        min="2"
        max="5"
      />
    );
    const antdFormItemMount = getAntdFormItemMount(wrapper,"password");
    expect(antdFormItemMount.childAt(0).childAt(0).props().className).toEqual("ant-form-item-control has-error");
  })
  it("min='2' && max='5' && value='1' should show error state",function(){
    const wrapper = mount(
      <Password 
        name="test"
        onlyLetterAndNumber={ false }
        value="1"
        min="2"
        max="5"
      />
    );
    const antdFormItemMount = getAntdFormItemMount(wrapper,"password");
    expect(antdFormItemMount.childAt(0).childAt(0).props().className).toEqual("ant-form-item-control has-error");
  })
  it("onlyLetterAndNumber=true should show success state",function(){
    const wrapper = mount(
      <Password 
        name="test"
        value="sddd1111"
        onlyLetterAndNumber={ true }
      />
    );
    const antdFormItemMount = getAntdFormItemMount(wrapper,"password");
    expect(antdFormItemMount.childAt(0).childAt(0).props().className).toEqual("ant-form-item-control has-success");
  })
  it("onlyLetterAndNumber=true should show error state",function(){
    const wrapper = mount(
      <Password 
        name="test"
        value="ddddd"
        onlyLetterAndNumber={ true }
      />
    );
    const antdFormItemMount = getAntdFormItemMount(wrapper,"password");
    expect(antdFormItemMount.childAt(0).childAt(0).props().className).toEqual("ant-form-item-control has-error");
  })
  it("rePassword=true should work",function(){
    const wrapper = mount(
      <Password 
        name="test"
        value="ddddd"
        onlyLetterAndNumber={ true }
        rePassword={ true }
      />
    );
    const antdRePasswordFormItemMount = wrapper.childAt(1).childAt(0).childAt(0).childAt(0);
    //console.log(antdRePasswordFormItemMount.props())
    expect(antdRePasswordFormItemMount.props().title).toEqual("重复密码");
  })
})

