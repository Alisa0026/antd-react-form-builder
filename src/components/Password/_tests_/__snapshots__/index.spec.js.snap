// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password renders correctly 1`] = `
<span
  class="password-con"
>
  <span>
    <div
      class="ant-row ant-form-item ant-form-item-with-help"
    >
      <div
        class="ant-form-item-label"
      >
        <label
          class="ant-form-item-required"
          title="name"
        >
          name
        </label>
      </div>
      <div
        class="ant-form-item-control-wrapper"
      >
        <div
          class="ant-form-item-control has-error"
        >
          <input
            class="ant-input test"
            name="test"
            type="password"
            value="55"
          />
          <div
            class="ant-form-explain"
          >
            必须介于5到12个字符，密码必须是字母和数字结合
          </div>
        </div>
      </div>
    </div>
  </span>
  <span>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-form-item-label"
      >
        <label
          class=""
          title="重复密码"
        >
          重复密码
        </label>
      </div>
      <div
        class="ant-form-item-control-wrapper"
      >
        <div
          class="ant-form-item-control "
        >
          <input
            class="ant-input test"
            name="re-test"
            type="password"
            value=""
          />
        </div>
      </div>
    </div>
  </span>
</span>
`;
